From: Aurelien Jarno <aurelien@aurel32.net>
Date: Sat, 19 Mar 2011 13:29:29 +0100
To: libc-ports@sourceware.org
Subject: [PATCH][alpha] Fix ____longjmp_chk

____longjmp_chk is broken on alpha due to a wrong register used for a
comparison. The patch below fixes that and with it the following
testsuite errors:
- tst-longjmp_chk.out, Error 1
- tst-chk2.out, Error 127
- tst-chk3.out, Error 127
- tst-chk5.out, Error 127
- tst-chk6.out, Error 127
- tst-lfschk2.out, Error 127
- tst-lfschk3.out, Error 127
- tst-lfschk5.out, Error 127
- tst-lfschk6.out, Error 127

2011-03-20  Aurelien Jarno  <aurelien@aurel32.net>

	* sysdeps/unix/sysv/linux/alpha/____longjmp_chk.S(____longjmp_chk):
	Fix wrong register in stack pointer comparison.

diff --git a/sysdeps/unix/sysv/linux/alpha/____longjmp_chk.S b/sysdeps/unix/sysv/linux/alpha/____longjmp_chk.S
index 5194ed8..e0e7106 100644
--- ports/sysdeps/unix/sysv/linux/alpha/____longjmp_chk.S
+++ ports/sysdeps/unix/sysv/linux/alpha/____longjmp_chk.S
@@ -58,7 +58,7 @@ ____longjmp_chk:
 	PTR_DEMANGLE2(s3, t1)
 	PTR_DEMANGLE2(fp, t1)
 #endif
-	cmpule	t0, sp, t1
+	cmpule	s3, sp, t1
 	bne	t1, .Lfail
 
 	.align	4
